server/src/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -134,7 +134,10 @@ public PluggableTask toPluggableTask(CRPluggableTask pluggableTask) {
 
     private void setCommonTaskMembers(AbstractTask task, CRTask crTask) {
         CRTask crTaskOnCancel = crTask.getOnCancel();
-        task.setCancelTask(crTaskOnCancel != null ? toAbstractTask(crTaskOnCancel) : null);
+        if(crTaskOnCancel != null)
```

Comment:
@tomzo - does it make sense to call the setter only if `crTask.getOnCancel()` is not null. The only reason being, if we ever decide to change the default value of `AbstractTask.onCancelConfig`, then there wont be multiple places to make the fix.